Dockerfile

```diff
@@ -0,0 +1,97 @@
+# ------------------------------------------------------------------------------
+# Build environment
+# ------------------------------------------------------------------------------
+    FROM rust:1.86-bookworm AS builder
+
+    # Install build dependencies
+    RUN apt-get update && apt-get install -y --no-install-recommends \
+        pkg-config \
+        libssl-dev \
+        && rm -rf /var/lib/apt/lists/*
+    
+    WORKDIR /app
+    
+    # Copy manifests first for better layer caching
+    COPY Cargo.toml Cargo.lock ./
+    COPY crates/solver-account/Cargo.toml crates/solver-account/
+    COPY crates/solver-config/Cargo.toml crates/solver-config/
+    COPY crates/solver-core/Cargo.toml crates/solver-core/
+    COPY crates/solver-delivery/Cargo.toml crates/solver-delivery/
+    COPY crates/solver-demo/Cargo.toml crates/solver-demo/
+    COPY crates/solver-discovery/Cargo.toml crates/solver-discovery/
+    COPY crates/solver-order/Cargo.toml crates/solver-order/
+    COPY crates/solver-pricing/Cargo.toml crates/solver-pricing/
+    COPY crates/solver-service/Cargo.toml crates/solver-service/
```

Comment:
This looks complicated to keep track of, we probably should cargo-chef at some point to copy all these manifests. Don't have to be in this PR